src/arcaflow_plugin_sdk/schema.py

```diff
@@ -116,6 +117,19 @@ def __str__(self):
         return "No such step: %s" % self.step
 
 
+@dataclass
+class NoSuchSignalException(Exception):
+    """
+    ``NoSuchSignalException`` indicates that the given signal is not supported by the plugin's step.
+    """
+
+    step: str
+    signal: str
+
+    def __str__(self):
+        return "No such signal '%s' in step '%s'" % self.signal, self.step
```

Comment:
Good points. Some of this was copy -> paste -> modify from code I didn't write. Might be worth a refactor.

Reply:
Yeah, in Pbench, we started down the path of customizing `Exception` classes, and I think we did it "improperly" (although, we didn't use `@dataclass` 😆), and after that we cookie-cutter'd them and now the blight is _everywhere_...we are correcting it as we go, but, until we have it nearly addressed, the odds are that a future developer will propagate the blight instead of propagating the correct practice.  🙁